docs/testing.md

```diff
@@ -0,0 +1,148 @@
+# Testing Using Docker
+
+**Table of Contents**
+
+- [Installation](#Installation)
+  - [Docker configuration](#docker-configuration)
+  - [Python configuration](#python-configuration)
+- [Basic Docker commands](#basic-docker-commands)    
+
+
+## Installation
+
+#### Docker configuration
+
+The base Docker installation methods for popular Linux distributions are listed here:
+	https://docs.docker.com/engine/install/
+
+Docker is not supported by **Fedora 33** officially. On my host system I used the following method (Moby based approach):
+
+1. Delete the old version of docker (if it was installed)
+
+   ```bash
+   sudo dnf remove docker-*
+   sudo dnf config-manager --disable docker-*
+   ```
+
+2. As docker doesn't support **CGroups2**, enable old **CGroups**
+
+   ```bash
+   sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+   ```
+
+3. Configure firewall
+
+   ```bash
+   sudo firewall-cmd --permanent --zone=trusted --add-interface=docker0
+   sudo firewall-cmd --permanent --zone=FedoraWorkstation --add-masquerade
+   ```
+
+4. Install Moby (an open-source project created by Docker)
+
+   ```bash
+   sudo dnf install moby-engine docker-compose
+   sudo systemctl enable docker
+   ```
+
+5. Add your user to **docker** group to run docker without **sudo**
+
+   ```bash
+   sudo groupadd docker
+   sudo usermod -aG docker $USER
+   ```
+
+6. Restart your computer
+
+   ```bash
+   sudo reboot
+   ```
+
+7. To test your installation you need running the following command
+
+   ```bash
+   sudo docker run hello-world
+   ```
+
+#### Python configuration
+
+- Install `python3.6` package in system using your distributive package manager tool: **apt**, **dnf**, **yum**, etc.
+
+- Clone/Update the `cortx-prvsnr` repository
+
+- Create the Python virtual environment using `venv` module
+
+  ```bash
+  cd /path/to/cortx-prvsnr
+  python3.6 -m venv ./venv
+  ```
+
+  :warning: NOTE: For more details about Python virtual environment see
+  https://docs.python.org/3.6/library/venv.html
+
+- Install provisioner API to `venv`: 
+
+  ```bash
+  pip install -e api/python[test]
+  ```
+
+- Start the docker environment creation
+
+  ```bash
+  pytest test/build_helpers -k test_build_setup_env -s --root-passwd root --nodes-num 1
```

Comment:
Could you please also make a note that it's possible to provide custom command line args using. Makes sense to copy-paste that part from #926 description.

Reply:
Done